build-tools/packages/build-cli/test/commands/typetests.test.ts

```diff
@@ -2,11 +2,330 @@
+				expect(pkg.devDependencies?.["test-package-previous"]).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it(`packageWithTypeValidation`, () => {
+				const pkg = clonedeep(packageWithTypeValidation);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.empty;
+				expect(pkg.typeValidation?.disabled).to.be.false;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it(`packageWithTypeValidationDisabled`, () => {
+				const pkg = clonedeep(packageWithTypeValidationDisabled);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.empty;
+				expect(pkg.typeValidation?.disabled).to.be.true;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("previousAndResetFalse", () => {
+			const name = "previousAndResetFalse";
+			const options = optionsMatrix.get(name)!;
+			it(`packageMinimal`, () => {
+				const pkg = clonedeep(packageMinimal);
+				updateTypeTestConfiguration(pkg, options);
+
+				expect(pkg.typeValidation).to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.be.undefined;
+				expect(pkg.typeValidation?.disabled).to.be.undefined;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it(`packageWithTypeValidation`, () => {
+				const pkg = clonedeep(packageWithTypeValidation);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).not.to.be.empty;
+				expect(pkg.typeValidation?.disabled).to.be.false;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.exist;
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it(`packageWithTypeValidationDisabled`, () => {
+				const pkg = clonedeep(packageWithTypeValidationDisabled);
+				updateTypeTestConfiguration(pkg, options);
+				expect(pkg.typeValidation).not.to.be.undefined;
+				expect(pkg.typeValidation?.broken).to.exist;
+				expect(pkg.typeValidation?.broken["broken-package"]).to.exist;
+				expect(pkg.typeValidation?.disabled).to.be.true;
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("clearAndResetTrue", () => {
+			const name = "clearAndResetTrue";
+			const options = optionsMatrix.get(name)!;
+			it(`packageMinimal`, () => {
+				const pkg = clonedeep(packageMinimal);
```

Comment:
rathert than using a clone function, if you make `packageMinimal` into a factory function instead of a constant, you can get a fresh opbject each type, and have more confidence no test will accidently modify it.

Reply:
Great idea.